src/fmu/dataio/providers/objectdata/_base.py

```diff
@@ -178,16 +184,72 @@ class ObjectDataProvider(ABC):
     time0: datetime | None = field(default=None)
     time1: datetime | None = field(default=None)
 
-    @staticmethod
-    def _validate_get_ext(fmt: str, subtype: str, validator: dict[str, V]) -> V:
-        """Validate that fmt (file format) matches data and return legal extension."""
-        try:
-            return validator[fmt]
-        except KeyError:
-            raise ConfigurationError(
-                f"The file format {fmt} is not supported. ",
-                f"Valid {subtype} formats are: {list(validator.keys())}",
+    def __post_init__(self) -> None:
+        """Main function here, will populate the metadata block for 'data'."""
+
+        # Don't re-initialize data if it's coming from pre-existing metadata.
+        if self.metadata:
+            return
```

Comment:
This is just the same `derive_metadata()` method that existed, with this extra condition, and the short form `meta = self.metadata` removed for the explicitness. No other changes

Reply:
This will be removed with the changes to preprocessed data, of course..